Lib/shutil.py

```diff
@@ -1415,7 +1418,7 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
             path.insert(0, curdir)
 
         # PATHEXT is necessary to check on Windows.
-        pathext = os.environ.get("PATHEXT", "").split(os.pathsep)
+        pathext = [ext for ext in os.environ.get("PATHEXT", _WIN_DEFAULT_PATHEXT).split(os.pathsep) if ext]
```

Comment:
Previously we'd get `[""]` for an empty variable, but now it will be `[]`. This will impact `files` below.

As I suggested above, I think in the case where we have no extensions we should use the provided name directly.

Reply:
@zooba So should I remove the again `_WIN_DEFAULT_PATHEXT` and return to `os.environ.get("PATHEXT", "")`? :)

On the second comment: would it make sense to tweak the `if` like this?

```python
if not pathext or any(cmd.lower().endswith(ext.lower()) for ext in pathext):
    files = [cmd]
else:
    files = [cmd + ext for ext in pathext]
```

Thank you very much in advance :)